transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/logging/BasicLoginModuleWithMultiplePrincipalsOnTheSubject.java

```diff
@@ -0,0 +1,170 @@
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wseb.logging;
+
+import java.security.Principal;
+import java.util.Map;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.kaazing.gateway.security.auth.config.parse.DefaultUserConfig;
+
+
+public class BasicLoginModuleWithMultiplePrincipalsOnTheSubject implements LoginModule {
+
+    private static final String TEST_PRINCIPAL_PASS = "testPrincipalPass";
+    private static final String TEST_PRINCIPAL_NAME = "testPrincipalName";
+    private DefaultUserConfig defaultPrincipal = new DefaultUserConfig();
+
+    // initial state
+    private Subject subject;
+    private Map<String, ?> sharedState;
+
+    // the authentication status
+    private boolean succeeded;
+    private boolean commitSucceeded;
+
+    // testUser's RolePrincipal
+    private RolePrincipal userPrincipal;
+
+    public void initialize(Subject subject,
+                           CallbackHandler callbackHandler,
+                           Map<String, ?> sharedState,
+                           Map<String, ?> options) {
+        this.subject = subject;
+        this.sharedState = sharedState;
+    }
+
+
+    @Override
+    public boolean login() throws LoginException {
+        // verify the username/password
+        String username = (String) sharedState.get("javax.security.auth.login.name");
+        char[] password = (char[]) sharedState.get("javax.security.auth.login.password");
+        if (username == null || password == null) {
+            throw new FailedLoginException("No UserName/Password to authenticate");
+        }
+        if (username.equals("joe") && password.length == 7 && password[0] == 'w' && password[1] == 'e'
+                && password[2] == 'l' && password[3] == 'c' && password[4] == 'o' && password[5] == 'm'
+                && password[6] == 'e') {
+            // authentication succeeded!!!
+            succeeded = true;
+            return true;
+        } else {
+            // authentication failed
+            succeeded = false;
+            throw new FailedLoginException("UserName/Password is Incorrect");
+        }
+    }
+
+    @Override
+    public boolean commit() throws LoginException {
+        if (!succeeded) {
+            return false;
+        } else {
+            // add a Principal (authenticated identity) to the Subject
+            userPrincipal = new RolePrincipal("USER");
+            subject.getPrincipals().add(userPrincipal);
+            commitSucceeded = true;
+
+            defaultPrincipal.setName(TEST_PRINCIPAL_NAME);
+            defaultPrincipal.setPassword(TEST_PRINCIPAL_PASS);
+            subject.getPrincipals().add(defaultPrincipal);
+
+            DefaultUserConfig defaultPrincipal2 = new DefaultUserConfig();
```

Comment:
Perhaps we can rename defaultPrincipal2 to defaultPrincipalNotForIdentityLogging or some similar type of name (this is just a suggestion in order to avoid using indexes for this type of variable)